pkg/controller/elasticsearch/client/v6.go

```diff
@@ -112,6 +113,24 @@ func (c *clientV6) DeleteVotingConfigExclusions(ctx context.Context, waitForRemo
 	return errors.New("Not supported in Elasticsearch 6.x")
 }
 
+func (c *clientV6) ClusterBootstrappedForZen2(ctx context.Context) (bool, error) {
+	// Look at the current master node of the cluster: if it's running version 7.x.x or above,
+	// the cluster has been bootstrapped.
+	// Even though c is a clientV6, it may be targeting a mixed v6/v7 having a v7 master.
+	var response Nodes
+	if err := c.get(ctx, "/_nodes/_master", &response); err != nil {
+		return false, err
+	}
+	if len(response.Nodes) != 1 {
+		return false, fmt.Errorf("GET /_nodes/_master returned %d nodes", len(response.Nodes))
```

Comment:
I'm trying to understand if returning an error is the right thing to do here. If there's no master it could be because election did not happen yet, imho it's not an error from the operator POV. My fear is that the error here prevents a downscale or an upgrade of the nodes.

Reply:
If there's no master I'm pretty sure we will not be able to downscale or upgrade nodes anyway? Also I think we don't get an HTTP response 200 to the call, but 503 if there's no master.
I think it's OK to leave as is?

Reply:
If there's no master then `GET /_nodes` doesn't respond immediately, but if it times out while waiting for the master (default `30s` I think) then it returns 503.

Reply:
`GET _nodes/_master` isn't a master-node action, it uses the node's local cluster state to work out which nodes are involved. This means it returns `200 OK` (but no nodes) if there is no master.

Reply:
> we will not be able to downscale or upgrade nodes anyway

Regarding the upgrade I think we should. That said, today, if there is no master (because of a bug or a bad configuration) user will have to delete Pods manually today because `ClearVotingConfigExclusions` will fail _(I think I will open an issue at least to document it)_

Reply:
Thanks for chiming in @DaveCTurner. Seems better to not return an error if the list of nodes is empty as you suggested @barkbay 👍 
I'm doing the change.